ext/json/ext/parser/parser.h

```diff
@@ -17,7 +17,9 @@
 # define MAYBE_UNUSED(x) x
 #endif
 
-#define option_given_p(opts, key) RTEST(rb_funcall(opts, i_key_p, 1, key))
+extern VALUE rb_hash_has_key(VALUE hash, VALUE key);
+
+#define hash_option_given_p(opts, key) (RTEST(rb_hash_has_key(opts, key)))
```

Comment:
Does this name need to be changed?

Reply:
Is there a problem with using an private API?

Reply:
Hi @Watson1978 I wasn't aware of your PR, I should have checked the other PRs more thoroughly before coding up my solution. Yours is better IMO with using only public API, and also making changes to `JSON.generate`. The only thing that my PR has that's different in the initialize function is that it checks if the hash is empty before going through all the option checks, which yours could add if you want. I also think your other PR (https://github.com/flori/json/pull/454) could be merged with the changes in https://github.com/flori/json/pull/481.

I'm fine with closing my PR now if yours gets a second look.
I'll make a new PR soon for the changes that I made that are not in `initialize` function.

Reply:
> Is there a problem with using a non public API?

Yes, that is not respecting the API and will likely break. Also it would likely not work e.g. on truffleruby needlessly.

Reply:
Okay, fixed